src/main/kotlin/com/expediagroup/openworld/sdk/core/plugin/authentication/AuthenticationHookFactory.kt

```diff
@@ -37,20 +37,22 @@ internal object AuthenticationHookFactory : HookFactory<AuthenticationConfigurat
 
 private class AuthenticationHookBuilder(private val client: Client) : HookBuilder<AuthenticationConfiguration> {
     private val log = OpenWorldLoggerFactory.getLogger(javaClass)
-    private val isLock = atomic(false)
+    private val lock = atomic(false)
     private val authenticationStrategy = client.getAuthenticationStrategy()
 
     override fun build(configs: AuthenticationConfiguration) {
         val httpClient = client.httpClient
 
         httpClient.plugin(HttpSend).intercept { request ->
-            if (authenticationStrategy.isNotIdentityRequest(request) && authenticationStrategy.isTokenAboutToExpire()) {
-                log.info(TOKEN_EXPIRED)
-                if (!isLock.getAndSet(true)) {
-                    try {
-                        authenticationStrategy.renewToken()
-                    } finally {
-                        isLock.compareAndSet(expect = true, update = false)
+            if (authenticationStrategy.isNotIdentityRequest(request)) {
```

Comment:
How does this make any difference? Using `&&` should be better as it allows for using the short-circuit evaluation.

Reply:
The difference is that "assignNewToken" now gets called and the token is assigned regardless of the presence of any token renewals. I have added a bit more detailed description of the issue in the description of the story.